src/routers/user.ts

```diff
@@ -87,29 +85,31 @@ user.delete(
             .json({ message: `user with id ${id} marked as deleted` })
         : res.status(404).json({ message: `no users with id ${id}` });
     } catch (e) {
-      res
-        .status(500)
-        .json({ message: e instanceof Error ? e.message : 'Unknown Error' });
+      next(e)
     }
   }
 );
 
 user.get(
   '/users',
   userQuerySubstringLimitValidator,
-  async (req: ValidatedRequest<QuerySubstringLimitSchema>, res: Response) => {
+  methodLoggerMiddleware,
+  async (req: ValidatedRequest<QuerySubstringLimitSchema>, res: Response, next: NextFunction) => {
     try {
       const { loginsubstring } = req.query;
       const { limit } = req.query;
 
       const result = await userService.getUsers(loginsubstring, limit);
       res.status(200).json({ users: result });
     } catch (e) {
-      res
-        .status(500)
-        .json({ message: e instanceof Error ? e.message : 'Unknown Error' });
+      next(e)
     }
   }
 );
 
+user.get('/error', methodLoggerMiddleware, function() {
+  //throw new Error("I AM UNHANDLED EXEPTION>>>>>");
+  //new Promise((res,rej)=>{rej('I AM UNHANDLED PROMISE REJECTION>>>>>')});
```

Comment:
You still have to do anything with response here, if you don't throw anything, or request will probably be timeouted.

Reply:
it is just for testing purposes